### DISCUSSION TITLE
Possible to have 'default' data with Forms?

### DISCUSSION THREAD

**Discussion by netsemantics** [jcubic/jquery.terminal#885] [Q&A] [answered]
Hi folks - does anyone know if there is a way to default data into a set of form prompts? I'd like to store a good number of settings in a settings object and then be able to just skip through and change one or two values later.  TIA   Ken

**Comment by jcubic** [comment, marked as answer]
You can edit the forms.js file and add defaults. You can even contribute with the changes if you want.

**Reply by jcubic** [reply]
It's two lines of code:
```javascript
        form: async function(term, data, initalData = {}) {
            var result = initalData;
            var types = $.terminal.forms.types;
            while (data.length) {
                var item = data.shift();
                const {type, name} = item;
                if (type && types[type]) {
                    if (item.message) {
                        term.echo(item.message);
                    }
                    result[name] = await this[types[type]](term, item);
                }
            }
            return result;
        },
```

**Reply by netsemantics** [reply]
I did look at that code - and your answer is a good one! My js skills are very limited but I will absolutely give it a try. Thanks so much for this cool application.
